### PR TITLE
py-fluidfft and co: add new packages

### PR DESCRIPTION
**Pull request by paugier**
This PR adds Fluidfft https://pypi.org/project/fluidfft and related packages.

It works as expected for other Spack packages I'm working on (https://foss.heptapod.net/fluiddyn/fluiddyn/-/merge_requests/128).

It depends on https://github.com/spack/spack/pull/46234 and https://github.com/spack/spack/pull/46235.

I have a question related to build options. One can build Fluidfft with Meson options

```
pip install fluidfft -C setup-args=-Dnative=true
```

I guess this can be used with `config_settings`, but I don't know how to do it.

### CODE REVIEW

**Comment by wdconinc** [comment]
> I guess this can be used with `config_settings`, but I don't know how to do it.

This may be a useful example:
https://github.com/spack/spack/blob/afc9615abf413cbccf2633932d55ee0b314c223a/var/spack/repos/builtin/packages/py-matplotlib/package.py#L282-L293

**Comment by wdconinc** [comment]
I noticed some dependency versions missing, similar to #46234 and #46235. Instead of marking them with suggestions, I am hoping that as main developer you are likely aware of where they need to be added.

File: var/spack/repos/builtin/packages/py-fluidfft/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    variant("native", default=False, description="Compile with -march=native and -Ofast.")
+
+    with default_args(type=("build", "run")):
+        depends_on("python@3.9:")
+        depends_on("py-transonic@0.6.4:")
+
+    with default_args(type="build"):
+        depends_on("py-meson-python")
+        depends_on("py-pythran")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/py-fluidfft/package.py]: pythran>=0.9.7

File: var/spack/repos/builtin/packages/py-fluidfft/package.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    with default_args(type="run"):
+        depends_on("py-fluiddyn@0.2.3:")
+        depends_on("py-pyfftw")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/py-fluidfft/package.py]: pyfftw >= 0.10.4

File: var/spack/repos/builtin/packages/py-fluidfft-p3dfft/package.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        depends_on("python@3.9:")
+        depends_on("py-mpi4py")
+
+    with default_args(type="link"):
+        depends_on("p3dfft3")
+        depends_on("fftw")
+
+    with default_args(type="build"):
+        depends_on("py-meson-python")
+        depends_on("py-transonic")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/py-fluidfft-p3dfft/package.py]: transonic>=0.6.1

File: var/spack/repos/builtin/packages/py-fluidfft-builder/package.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    pypi = "fluidfft-builder/fluidfft_builder-0.0.2.tar.gz"
+
+    maintainers("paugier")
+    license("MIT", checked_by="paugier")
+
+    version("0.0.2", sha256="c0af9ceca27ae3a00ccf2f160703be9e394d8b886b8a02653b6c0a12a4f54a90")
+
+    depends_on("python@3.9:", type=("build", "run"))
+    depends_on("py-flit-core", type="build")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/py-fluidfft-builder/package.py]: flit_core >=3.2,<4

**Review by wdconinc — COMMENTED** [review]
A few more version additions. I think this is all that's left.

Please consider just making the change as a commit in the branch instead of rebasing with the new changes. We squash merge package updates so many commits in the branch are no problem. It also makes it easier to review only the additions. Finally, rebasing generally slows down merging due to our CI pipelines, as described in https://github.com/spack/spack-infrastructure/blob/main/docs/deferred_pipelines.md#tips.

File: var/spack/repos/builtin/packages/py-fluidfft-builder/package.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    pypi = "fluidfft-builder/fluidfft_builder-0.0.2.tar.gz"
+
+    maintainers("paugier")
+    license("MIT", checked_by="paugier")
+
+    version("0.0.2", sha256="c0af9ceca27ae3a00ccf2f160703be9e394d8b886b8a02653b6c0a12a4f54a90")
+
+    depends_on("python@3.9:", type=("build", "run"))
+    depends_on("py-flit-core@3.2:4", type="build")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/py-fluidfft-builder/package.py]: To exclude 4, we'd need:
```suggestion
    depends_on("py-flit-core@3.2:3", type="build")
```
Version ranges are not using the python convention.

**Review by wdconinc — APPROVED** [review]
Great! Ready to merge.

File: var/spack/repos/builtin/packages/py-fluidfft-builder/package.py (hunk 5 above)
Comment (paugier) [reply var/spack/repos/builtin/packages/py-fluidfft-builder/package.py]: Sorry to do so many mistakes and thanks a lot to help and explain. I corrected this range.

**Comment by paugier** [comment]
Thanks a lot @wdconinc. I start to understand better how Spack works and how I can contribute. This will be very convenient for some applications on clusters.

**Pull request merged**